sequence_processing_pipeline/Job.py

```diff
@@ -226,6 +226,9 @@ def wait_on_job_ids(self, job_ids, callback=None):
         # them before returning, optionally submitting callbacks for each
         # job-id.
 
+        # ensure all ids are strings to ensure proper working w/join().
+        job_ids = [str(x) for x in job_ids]
```

Comment:
What edge case caused problems here?

Reply:
I once passed in a list of job-ids that had been converted to int()s in the process of error checking and it made the entire job fail because of a downstream str.join(). IMHO this is prudent as there's nothing else checking the input parameters for this method and this kind of error is the kind that can be hard to debug for some down the road.  Moreover because it uses a list comprehension it should be pretty cheap.